service/history/historyEngine.go

```diff
@@ -2254,20 +2254,84 @@ type updateWorkflowAction struct {
 	createDecision bool
 }
 
-func (e *historyEngineImpl) updateWorkflowExecutionWithAction(
+type updateWorkflowActionFunc func(mutableState mutableState) (*updateWorkflowAction, error)
+
+func (e *historyEngineImpl) updateWorkflowExecutionWithActionAndContextReload(
```

Comment:
this function is confusing, in one path, it is a pass through; another path, release function is called to handled error cases.
the general idea of the release function is: that function should always be called in a defer function whenever seeing it in the first place, to guard against potential bug / panic